code/ObjFileMtlImporter.cpp

```diff
@@ -302,6 +302,8 @@ void ObjFileMtlImporter::createMaterial()
         // New Material created
         m_pModel->m_pCurrentMaterial = new ObjFile::Material();
         m_pModel->m_pCurrentMaterial->MaterialName.Set( name );
+        if (m_pModel->m_pCurrentMesh)
+            m_pModel->m_pCurrentMesh->m_uiMaterialIndex = m_pModel->m_MaterialLib.size();
```

Comment:
If the materialLib indices are zero based the index of the one you are adding will be materialLib.size() before it is added. If I set the material index AFTER adding the new material, then the index of that material would be materialLib.size() -1. I think if you use materialLib.size() -1 and you have not added anything yet you will get a negative index.